src/Core/Manifest/Module.php

```diff
@@ -275,6 +293,79 @@ public function hasResource($path)
             ->getResource($path)
             ->exists();
     }
+
+    /**
+     * Determine what CI library the module is using.
+     * @internal
+     */
+    public function getCILibrary(): string
```

Comment:
The 'CI Library' naming is confusing?  We know what CI is, lots of projects will use it, though the first point of failure for most devs may be "I cannot run unit-tests on my laptop", and 'CI library' doesn't really make sense here?

What are we testing here?  The PHP version of a module in the vendor folder?

Reply:
I agree with Steve that the method (and the use of it across the board) is confusing. Based on the code, we're trying to get a version of PHPUnit that the module requires.

Perhaps `getPHPUnitVersion`?

Reply:
My aim is to future proof this method a bit and make it technologically agnostic. 

I'm not trying to return an exact version. I'm trying to return a broad type of set up describing what CI set up the module is using. These broad types are going to be specific to Silverstripe. For example, if PHPUnit 10 works without any change, my instinct will be to update the method to return `CI_PHPUNIT_NINE` for it.

Maybe we'll make another substantial change to our CI setup beyond changing the PHPUnit version that forces us to treat modules differently. Or maybe one day, we'll decide to support a different CI library for running test. ([Codeception](https://codeception.com/) anyone?)

`getPHPUnitVersion` seems too specific, A reasonable person could look at this and expect an exact version number to be returned.

What about something like `getCIConfiguration` or `getCIModule` or `getCISetup`?

It's also worth pointing out that it's unlikely this will surface all the way to developers. If an exception is thrown it's not going to be coming from this method.

Reply:
I get the sentiment now (was on the track before, too), thanks for spelling it out again though @maxime-rainville!

A few more name suggestions, considering the context of the method is the vendor module — `getCIConfig` (CI specific config — do we want to be more generic?), `getModuleConfig` (general module config), `getModuleDeps` (dependencies, might be ambiguous) or `getModuleReqs` (requirements, but also could be seen as something else).

Also, how about returning an array of constants? That would be in line with future-proofing it to return a number of constants related to different aspects of the module. And then the consumer can cherry-pick the ones related to it or do an `in_array` test.

Reply:
May change it to getCIRequirements() and return an array instead of a string
then it can return `[PHPUNIT_9, BEHAT, PHPCS, NPM_VERSION]` and whatever else in the future

Reply:
Having a method that allows you to return additional info about the module testing set up could be interesting ... at the same time I'm also mindful about making this unduly complex for hypothetical use case we never need. 

It's pretty clear to me we'll have to deal with PHPUnit v10 one day ... it's not clear we'll ever have a need to know what JS CI the module is using. If that kind of need ever arose, my gut feeling would be to add an extra method for each aspect that is being tested, rather than use an array.

If we return an array, then each value should probably have a key that describes what aspect is being tested. e.g.:
```php
[
    'PHP' => self::CI_PHPUNIT_NINE,
    'JS' => self::CI_JEST,
    'Behaviour' => self::CI_BEHAT
];
```

Initially, only the PHP one would be returned.

`getCIConfig` is my favourite one right now. `getModuleConfig` or `getModuleDeps` could be a confusing since it's on a class called `Module` and this really is about your CI set up.

It's also worth remembering that this method will be marked `@internal`, so we can change our mind later.

Reply:
`getCIConfig()` seems fine given it's on `Module`

If you want to keep the return type as a string, then name the method getCIPhpSomething() - since it'll have to be constrained to php only otherwise it's way too messy

Reply:
Cool ... I'll go with `getCIConfig()` and an array return type.